frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.unit.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import { screen } from "@testing-library/react";
+import fetchMock from "fetch-mock";
+import _ from "underscore";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders } from "__support__/ui";
+import { Card, Collection } from "metabase-types/api";
+import {
+  createMockCard,
+  createMockNativeDatasetQuery,
+} from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import { createMockState } from "metabase-types/store/mocks";
+import { checkNotNull } from "metabase/core/utils/types";
+import { getMetadata } from "metabase/selectors/metadata";
+import NativeQuery from "metabase-lib/queries/NativeQuery";
+
+const { NativeQueryEditor } = jest.requireActual(
+  "metabase/query_builder/components/NativeQueryEditor",
+);
+
+const TEST_DB = createSampleDatabase();
+
+const TEST_NATIVE_CARD = createMockCard({
+  dataset_query: createMockNativeDatasetQuery({
+    database: TEST_DB.id,
+  }),
+});
+
+const ROOT_COLLECTION = {
```

Comment:
`= createMockCollection({ id: "root", ...})`. Please never cast invalid objects forcefully, this can become something very hard to migrate from.

Reply:
ee7c716bde4f5afb9bb9ae27a02faeb7d1c63e74